uavcan/node/432.GetPortInfo.1.0.uavcan

```diff
@@ -0,0 +1,26 @@
+#
+# This service is used to obtain information about a port (either message or service).
+# It can be used for advanced network inspections, e.g. for building computational graph maps.
+#
+
+# The port of interest; can be either a service or a message.
+void7
+PortID.1 port_id
+
+---
+
+# Flags describing the current port.
+# Note that a message port can be both subscriber and publisher at the same time.
+bool is_subscriber      # Message input port
```

Comment:
We should really create different services for subject ID and service ID instead. This is a bit messy

Reply:
We don't want to add more standard services than we really have to because they are taxing on the memory footprint -- you have to maintain a lot of state information to manage service requests. Seeing as this service serves only inspection needs and does not in any way help the node fulfill its utility function directly, node designers will be too keen to drop these if they find them too resource consuming or too difficult to implement.

Reply:
How would two separate services be more taxing for ram footprint than one? Wouldn't one service mean that it would have the sum of calls that both services had if they were separate and ram use end up more or less the same?

Reply:
See section 4.3.2 "State variables" of the current spec draft or [the old spec](https://uavcan.org/Specification/4._CAN_bus_transport_layer/#state-variables). We have a set of variables to maintain per every service we want to support, hence increased footprint.